src/server/acl/acl_family.cc

```diff
@@ -61,10 +67,108 @@ void AclFamily::List(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+namespace {
+
+std::optional<std::string> MaybeParsePassword(std::string_view command) {
+  if (command[0] != '>') {
+    return {};
+  }
+
+  return {std::string(command.substr(1))};
+}
+
+std::optional<bool> MaybeParseStatus(std::string_view command) {
+  if (command == "ON") {
+    return true;
+  }
+  if (command == "OFF") {
+    return false;
+  }
+  return {};
+}
+
+using OptCat = std::optional<uint32_t>;
+
+// bool == true if +
+// bool == false if -
+std::pair<OptCat, bool> MaybeParseAclCategory(std::string_view command) {
+  if (absl::StartsWith(command, "+@")) {
+    auto res = CATEGORY_INDEX_TABLE.find(command.substr(2));
+    if (res == CATEGORY_INDEX_TABLE.end()) {
+      return {};
+    }
+    return {res->second, true};
+  }
+
+  if (absl::StartsWith(command, "-@")) {
+    auto res = CATEGORY_INDEX_TABLE.find(command.substr(2));
+    if (res == CATEGORY_INDEX_TABLE.end()) {
+      return {};
+    }
+    return {res->second, false};
+  }
+
+  return {};
+}
+
+using facade::ErrorReply;
+
+std::variant<User::UpdateRequest, ErrorReply> ParseAclSetUser(CmdArgList args) {
+  User::UpdateRequest req;
+
+  for (auto arg : args) {
+    if (auto pass = MaybeParsePassword(facade::ToSV(arg)); pass) {
+      if (req.password) {
+        return ErrorReply("Only one password is allowed");
+      }
+      req.password = std::move(pass);
+      continue;
+    }
+
+    ToUpper(&arg);
+    const auto command = facade::ToSV(arg);
+
+    if (auto status = MaybeParseStatus(command); status) {
+      if (req.is_active) {
+        return ErrorReply("Multiple ON/OFF are not allowed");
+      }
+      req.is_active = *status;
+      continue;
+    }
+
+    auto [cat, add] = MaybeParseAclCategory(command);
+    if (!cat) {
+      return ErrorReply(absl::StrCat("Unrecognized paramter", command));
+    }
+
+    auto* acl_field = add ? &req.plus_acl_categories : &req.minus_acl_categories;
```

Comment:
This is from the previous PR. What do you mean `no` caregory? You mean `NONE` ?

Reply:
like what happens if I do `ACL SETUSER +@JSON -@JSON`

Reply:
So I thought about this before, and for this specific case there won't be an issue because the ADD set is applied Before the REMOVE set and I thought that this redundancy is fine. However, now I am thinking about it, for the reverse case this is not true, that is `-@JSON +@JSON` won't do what is expected and therefore I think we should only allow a group once per `SET USER` command (and probably this will require some special handling for some categories, example is ` -@JSON +@ALL`). 

Another solution is to apply them one by one in order which is slightly slower (since I can apply them all at once) but I don't expect it will affect the performance and IMO I think this is a better solution since it covers the ordering and we can allow `+@JSON -@JSON` and vice versa.

Reply:
I will address this in a separate PR since this is about the `AUTH` command

Reply:
Wait... What operations permitted? If the string is SSO, then any operations on the map (rehashing, shrinking, etc) will invalidate the string_view 🤨 

```
using RegistryType = absl::flat_hash_map<std::string, User>;
```